### PR TITLE
Out of memory check fix for tee_mm_alloc

### PR DESCRIPTION
**Pull request by youssefesmat**
Hello,

This is trying to address open issue #304.
I also restructured the one of the if statements to avoid overflow due to the size parameter.

Thanks,
Youssef

### CODE REVIEW

File: core/arch/arm/mm/tee_mm.c [hunk 1]
```diff
@@ -115,15 +117,22 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, uint32_t size)
 
 	/* check if we have enough memory */
 	if (pool->flags & TEE_MM_POOL_HI_ALLOC) {
-		if (((entry->offset << pool->shift) - pool->lo) < size)
+		/*
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/mm/tee_mm.c]: I think we are sure there is enough memory when `entry->next != NULL`. So we could test this before the test `if (pool->flags & TEE_MM_POOL_HI_ALLOC)`

**Comment by pascal-brand-st-dev** [comment]
Apart from a comment that could be or not taken into account:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
@youssefesmat Thank you very much for your contribution.

In order to have clean code, we are trying to remove all the minor steps towards a clean pull-request. Could you please follow  https://github.com/OP-TEE/optee_os/blob/master/documentation/github.md#finalizing-the-contribution so that we push your contribution on master.

**Comment by pascal-brand-st** [comment]
Slighly change the commit title, rebased on master, and merge manually.

**Pull request closed without merging**